llvm/lib/Analysis/IR2Vec.cpp

```diff
@@ -261,55 +262,121 @@ void FlowAwareEmbedder::computeEmbeddings(const BasicBlock &BB) const {
   BBVecMap[&BB] = BBVector;
 }
 
+// ==----------------------------------------------------------------------===//
+// VocabStorage
+//===----------------------------------------------------------------------===//
+
+VocabStorage::VocabStorage(std::vector<std::vector<Embedding>> &&SectionData)
+    : Sections(std::move(SectionData)) {
+  TotalSize = 0;
```

Comment:
you could make `TotalSize` and `Dimension` `const`. Then, just have a little static (or you can declare and call a lambda in place) and initialize at construction (i.e. 

```
TotalSize([&]() {...}()), ...
```

It's a tad convoluted, but readability will be great.